### ISSUE TITLE
Remove button not working on cart page

### ISSUE THREAD

**Issue by nodarik** [ColorlibHQ/Fashe#29]
add one item in the cart, then go to the cart page and try to remove item,
check this image:

https://colorlib.com/wp/wp-content/uploads/sites/2/screenshot-remove-button-not-working.png

**Comment by sicdigital** [comment]
@nodarik If you refresh the page was the item removed? I had to add some code for this to refresh after hitting the remove button.

**Comment by cristianraiber** [comment]
@sicdigital - create a PR with your changes, please? :)

**Comment by sicdigital** [comment]
Yeah, I can make a PR, I also made my version be able to go down to quantity 0, do you want me to include that. Currently you can only update the quantity of items in the cart to anything above 1.

**Comment by cristianraiber** [comment]
@sicdigital - include everything you've done and I'll get one of the developers to review your PR and merge accordingly :)

**Comment by lexvz** [comment]
Has this issue been solved? A client of mine has the same issue (:

**Comment by mtosalim** [comment]
Hello, dont know if anyone will need this in the future, but:
in cart.liquid ~line:78, remove {{ ‘cart.general.remove’ | t }}
and replace with < a href="/cart/change?line=1&amp;quantity=0"> Remover < /a >, dont forget to remove the spaces after < and before >.

**Comment by essarouri** [comment]
@mtsalim yes it's work thank you

**Comment by Smj90** [comment]
> Hello, dont know if anyone will need this in the future, but:
> in cart.liquid ~line:78, remove {{ ‘cart.general.remove’ | t }}
> and replace with < a href="/cart/change?line=1&quantity=0"> Remover < /a >, dont forget to remove the spaces after < and before >.

It Worked!!!! :)

**Comment by martentaavet** [comment]
> Hello, dont know if anyone will need this in the future, but: in cart.liquid ~line:78, remove {{ ‘cart.general.remove’ | t }} and replace with < a href="/cart/change?line=1&quantity=0"> Remover < /a >, dont forget to remove the spaces after < and before >.

the code provided before seems to delete the first product in the cart.

If anybody needs it i wrote this snippet you should replace this snippet of the code:

< button class="remove s-text7 flex-c-m m-l-30 m-t-10 p-l-10 p-r-10 p-t-5 p-b-5 bo1 bo-rad-23 hov1 trans-0-4" data-key="{{ item.variant_id }}" >
{{ 'cart.general.remove' | t }}
< /button >

with this snippet:

< form action="/cart/change" method="post" id="cart_form_{{ item.id }}" >
  < input type="hidden" name="line" value="{{ forloop.index }}" >
  < input type="hidden" name="quantity" value="0 ">
  < button type="submit" class="remove s-text7 flex-c-m m-l-30 m-t-10 p-l-10 p-r-10 p-t-5 p-b-5 bo1 bo-rad-23 hov1 trans-0-4" >
    Remove
  < /button >
< /form >


also as said before then dont forget to remove the spaces after < and before >.

**Comment by puikinsh** [comment]
Fashe 2.0 replaced the cart entirely, so the code this report is against no longer exists.

The v1 cart was jQuery driving a bespoke ajax layer in `assets/common.js.liquid`, which is where nearly every cart report on this repo originated. 2.0 uses a `cart-drawer` custom element on Shopify's Cart AJAX API:

- removing a line is `/cart/change` with `quantity: 0` — no page navigation
- quantity changes fire on the input's `change` event, so there is no "Update cart" button that can send you to checkout
- adding to cart posts `FormData` and re-renders the cart drawer and the header cart bubble through the Section Rendering API, instead of navigating to `/cart` or leaving a stale empty cart behind

Closing as resolved by the rewrite.

This is based on reading the 2.0 code rather than reproducing your original report, so if you are on 2.0.1 and still hit it, please reopen with the steps.

2.0.1: https://github.com/ColorlibHQ/Fashe/releases/tag/v2.0.1
